src/main/java/Model/Person.java

```diff
@@ -0,0 +1,46 @@
+package Model;
+
+import javax.persistence.*;
+
+@Entity
+public abstract class Person {
+
+    @Id
+    @GeneratedValue(strategy= GenerationType.AUTO)
+    private Integer id = null;
+    private String name = null;
+    private String type = null;
+
+    public Person() {}
+
+    public Person(String name){
+        this.name = name;
+    }
+    public Person(String name, String type){
+        this.name = name;
+        this.type = type;
+    }
+
+    public Integer getId(){
+        return this.id;
+    }
+
+    public String getName(){
+        return this.name;
+    }
+
+    public void setName(String name){
+        this.name = name;
+    }
+
+    public String getType(){
+        return this.type;
+    }
+
+    @Override
+    public String toString() {
+        return "Person [id=" + id + ", name=" + name + ", type=" + type +"]";
```

Comment:
Probably don't need to implement as it is an abstract class.

Reply:
Also what would happen if the values were null? Would it just print nothing?

Reply:
The toString could be removed.
We are doing an abstract class because it's better to do that instead of just making it a superclass.